test/integration/targets/aws_ses_identity_dkim/tasks/main.yml

```diff
@@ -0,0 +1,128 @@
+      aws_ses_identity:
+        identity: "{{ dkim_identity_domain }}"
+        state: present
+        <<: *aws_connection_info
+      register: ses_domain_identity
+
+    - name: Create SES identity for running DKIM module tests on (email identity)
+      aws_ses_identity:
+        identity: "{{ dkim_identity_email }}"
+        state: present
+        <<: *aws_connection_info
+      register: ses_email_identity
+
+    ## DKIM enable test
+
+    - name: Enable DKIM verification on domain identity
+      aws_ses_identity_dkim:
+        identity: "{{ dkim_identity_domain }}"
+        state: enabled
+        <<: *aws_connection_info
+      register: domain_result
+
+    - name: assert successful result
+      assert:
+        that:
+          - domain_result.changed == True
+          - domain_result.dkim_enabled == True
+          - domain_result.dkim_tokens|length > 0
+
+    - name: Enable DKIM verification on email identity
+      aws_ses_identity_dkim:
+        identity: "{{ dkim_identity_email }}"
+        state: enabled
+        <<: *aws_connection_info
+      register: email_result
+
+    - name: assert successful result
+      assert:
+        that:
+          - email_result.changed == True
+          - email_result.dkim_enabled == True
+          - email_result.dkim_tokens|length > 0
+
+    ## DKIM idempotence test (rerun)
+
+    - name: Enable DKIM verification on domain identity (rerun)
+      aws_ses_identity_dkim:
+        identity: "{{ dkim_identity_domain }}"
+        state: enabled
+        <<: *aws_connection_info
+      register: domain_result
+
+    - name: assert no change (itempotence)
+      assert:
+        that:
+          - domain_result.changed == False
+          - domain_result.dkim_enabled == True
+
+    - name: Enable DKIM verification on email identity (rerun)
+      aws_ses_identity_dkim:
+        identity: "{{ dkim_identity_email }}"
+        state: enabled
+        <<: *aws_connection_info
+      register: email_result
+
+    - name: assert no change (idempotence)
+      assert:
+        that:
+          - email_result.changed == False
+          - email_result.dkim_enabled == True
+
+    ## DKIM disable test
+
+    - name: Disable DKIM verification on domain identity
+      aws_ses_identity_dkim:
+        identity: "{{ dkim_identity_domain }}"
+        state: disabled
+        <<: *aws_connection_info
+      register: domain_result
+
+    - name: assert result changed
+      assert:
+        that:
+          - domain_result.changed == True
+          - domain_result.dkim_enabled == False
+
+    - name: Disable DKIM verification on email identity
+      aws_ses_identity_dkim:
+        identity: "{{ dkim_identity_email }}"
+        state: disabled
+        <<: *aws_connection_info
+      register: email_result
+
+    - name: assert result changed
+      assert:
+        that:
+          - email_result.changed == True
+          - email_result.dkim_enabled == False
+
```

Comment:
It would be good to add a couple of tests for check_mode as well to make sure it doesn't actually get broken.

There's an example of how to do this here https://github.com/ansible/ansible/blob/devel/test/integration/targets/aws_ses_identity/tasks/main.yaml#L151

Essentially, you can specify

    check_mode: True

On the task to force a task into check mode, then of course you're asserting that it didn't change anything but reports that it did.

Reply:
Done